test/partitioning.jl

```diff
@@ -85,12 +85,27 @@
     p = partition(grid, BlockPartition(T(5),T(2)))
     @test length(p) == 12
     @test Set(nelements.(p)) == Set([5,10])
+
+    grid = CartesianGrid{T}(50, 50, 50)
+
+    p = partition(grid, BlockPartition(T(1.), T(1.), T(1.), neighbors = false))
+    @test length(p) == 125000
+    @test Set(nelements.(p)) == Set(1)
+    @test metadata(p) == Dict{Any,Any}()
+
+    p = partition(grid, BlockPartition(T(5.), T(5.), T(5.), neighbors = true))
+    @test length(p) == 1000
+    @test Set(nelements.(p)) == Set(125)
+    n = metadata(p)[:neighbors]
+    @test length(n) == length(p)
+    @test all(0 .< length.(n) .< 27) 
+    
   end
 
   @testset "BisectPointPartition" begin
     grid = CartesianGrid((10,10), T.((-0.5,-0.5)), T.((1.0, 1.0)))
 
-    p = partition(grid, BisectPointPartition(T.((0.0,1.0)), T.((5.0,5.1))))
+    p = partition(grid, BisectPointPartition(T.((0.0,1.0)), T.((5.0,5.1)))) 
```

Comment:
```suggestion
    p = partition(grid, BisectPointPartition(T.((0.0,1.0)), T.((5.0,5.1))))
```
Please be careful with empty spaces

Reply:
:+1: